src/lib/utils/synapseTypes/Principal/PrincipalServices.ts

```diff
@@ -0,0 +1,53 @@
+// Principal Services
+// http://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.PrincipalController
+
+
+// http://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/principal/AliasCheckRequest.html
+export type AliasCheckRequest = {
+  alias: string
+  type: AliasType
+}
+
+// http://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/principal/AliasType.html
+export enum AliasType {
+  USER_NAME = "USER_NAME",
+  TEAM_NAME = "TEAM_NAME",
+  USER_EMAIL = "USER_EMAIL",
+  USER_OPEN_ID = "USER_OPEN_ID",
+  USER_ORCID = "USER_ORCID"
+}
+
+// http://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/principal/AliasCheckResponse.html
+export type AliasCheckResponse = {
+  available: boolean //True is the alias is available.
+  valid: boolean //True if the alias is valid for the passed type.
+}
+
+// http://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/auth/NewUser.html
+export type NewUser = {
+  email: string
+  userName?: string
+  firstName?: string
+  lastName?: string
+  encodedMembershipInvtnSignedToken?: string
+}
+
+// http://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/principal/AccountSetupInfo.html
+export type AccountSetupInfo = {
+  username: string
+  password: string
+  firstName?: string
+  lastName?: string
+  emailValidationSignedToken?: EmailValidationSignedToken
+}
+
```

Comment:
I don't know if it makes sense, unless we add other similar type, but may be have a base type of 
```AccountBase = {
  firstName?: string
  lastName?: string
userName?: string
}`` and have AccountSetupInfo and NewUser extend it. Also, I think the capitalization of useName vs username should be consistent.

Reply:
Yes, but these are not my class type definitions, they are defined by the backend.
http://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/principal/AccountSetupInfo.html
http://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/auth/NewUser.html
While I would love for the capitalization to match, it does not match in the API, so <shrug>